src/Compilers/CSharp/Portable/Symbols/SymbolCompletionState.cs

```diff
@@ -34,16 +34,35 @@ internal int IncompleteParts
         }
 
         /// <summary>
-        /// Used to force (source) symbols to a given state of completion.
+        /// Used to force (source) symbols to a given state of completion when the only potential remaining 
+        /// part is attributes. This does force the invariant on the caller that the implementation of 
+        /// of <see cref="Symbol.GetAttributes"/> will set the part <see cref="CompletionPart.Attributes"/> on
+        /// the thread that actually completes the loading of attributes. Failure to do so will potentially
+        /// result in a deadlock.
```

Comment:
There is also no-reentrancy requirement. - If the `GetAttributes` thread somehow ends up here before setting the flag, it would wait for itself.

Reply:
The part where GetAttributes may complete without setting/waiting for the completion flag feels unfortunate. Really nothing can be done about that?

The behavior contracts that we have to assume, even if they hold, seem a bit accidental.

Reply:
I mean the fix looks ok as a fix, but perhaps there is a need for a follow up with the root problem - to fix `GetAttributes` completion or to prove it is not possible/desirable to fix?